object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -148,25 +152,27 @@ private void handleInvocation(Set<LocalVarWithTree> newDefs, Node node) {
    * {@code @MustCall} type, then its result is pseudo-assigned to some location that can take
    * ownership of the result
    */
-  private void checkPseudoAssignToOwning(Node node) {
+  private void checkPseudoAssignToOwning(Node node, Set<Set<LocalVarWithTree>> defs) {
     Tree tree = node.getTree();
     List<String> mustCallVal = typeFactory.getMustCallValue(tree);
     if (mustCallVal.isEmpty()) {
       return;
     }
-    boolean assignedToOwning = false;
+    boolean assignedToOwningOrMustCallChoice = false;
     AssignmentContext assignmentContext = node.getAssignmentContext();
     if (assignmentContext != null) {
       Element elementForType = assignmentContext.getElementForType();
       if (assignmentContext instanceof AssignmentLhsContext) {
         // lhs should be a local variable
-        assignedToOwning = isOwningAssignmentLhs(elementForType);
+        assignedToOwningOrMustCallChoice = isOwningAssignmentLhs(elementForType);
       } else if (assignmentContext instanceof MethodParameterContext) {
         // must be an @Owning parameter
-        assignedToOwning = typeFactory.getDeclAnnotation(elementForType, Owning.class) != null;
+        assignedToOwningOrMustCallChoice =
+            typeFactory.getDeclAnnotation(elementForType, Owning.class) != null
+                || typeFactory.hasMustCallChoice(elementForType);
```

Comment:
So the idea here is if the formal parameter is `@MustCallChoice`, we "pretend" the call was assigned to an owning location, since we can satisfy the `@MustCall` obligation on the returned value from the call; is that right?

Reply:
exactly!

Reply:
I think it's better to keep this check here because we need to know if the method call or object creation node is passed to a @MustCallChoice parameter or not! So, with AssignmentContext we can do this check! what do you think about changing the name of checkPseudoAssignToOwning method instead?!

Reply:
Sorry I wrote this comment in the wrong place!  Nevermind 🙂